src/pages/docs/guides/chat/bring-own-database.mdx

```diff
@@ -0,0 +1,123 @@
+2. Using [outbound streaming](/docs/platform/integrations/streaming).
+3. Using an [Ably queue](/docs/platform/integrations/queues).
+4. Publishing via your own servers.
+
+## Decoding and storing messages
+
+Regardless of the delivery mechanism, you will need to decode the received messages into Chat messages. Details of the mapping from Ably Pub/Sub messages to Chat messages are available in the [Integrations](/docs/chat/integrations) documentation.
+
+After performing the decoding and you have a chat `Message` object, you can:
+
+1. Save it to your own database, index by `serial`.
+2. If the message already exists by `serial`, it means you have received an update, delete, or reaction summary update. To check if you need to update the message, you can use the `version.serial` to compare the latest version of the message you have received with the version of the message you have in your database. Lexicographically higher means newer version.
+3. If you want to store reaction summaries, always update the reactions field when receiving a reaction summary update (action `4` or `message.summary`).
+
+<Code>
+```typescript
+const saveOrUpdateMessage = (message: Message) => {
+  // Check if the message already exists by `serial`
+  const existingMessage = await getMessageBySerial(message.serial);
+  if (!existingMessage) {
+    // message not yet in your database => save it
+    await saveMessage(message);
+    return;
+  }
+  
+  if (message.version.serial < existingMessage.version.serial) {
+      // if received version is older, discard
+    return;
+  } else if (message.version.serial === existingMessage.version.serial && message.action !== 'message.summary') {
+    // if the message is the same version, and the action is not a summary, discard
+    return;
+  }
+
+  // message is newer or it's a summary event => update the message
+  await updateMessage(existingMessage, message);
+};
+```
+</Code>
+
+## Using a webhook via integration rules
+
+Ably can forward messages to your own system via a webhook. This is the simplest to setup if you don't already have other systems in place for message ingestion. This section covers the simple HTTP endpoint webhook, but the same principles apply to other webhook integrations such as AWS Lambda, Azure Function, Google Function, and others.
+
+Read the guide on [outbound webhooks](/docs/platform/integrations/webhooks) for more details on how to setup the webhook with Ably for the platform of your choice.
+
+All webhook integrations allow you to use a regex filter on the channel name to control which channels the webhook should be triggered for. To enable the webhook to trigger for all chat rooms, use the `.*::\$chat$` regex. All chat rooms are backed by a channel suffixed with `::$chat`. For chat room `foo` the channel name is `foo::$chat`.
```

Comment:
We're aiming to avoid talking about directives where we can. Perhaps a better way to approach this would be to suggest that people group their chat rooms in a common namespace `e.g. chat:<roomName>` and then apply the regex to `chat:*`? Then at least we only have to talk about chopping the ::$ off the end.

@m-hulbert WDYT?